databases/week2/hyf_lesson2HomeWork.sql

```diff
@@ -0,0 +1,144 @@
+-- PART 1: Working with tasks
+
+--  #1a: Add a task with these attributes: title, description, created, updated, due_date, status_id, user_id:
+INSERT INTO task (
+title, 
+description, 
+created, 
+updated, 
+due_date, 
+status_id)
+VALUES (
+'Midnight struggles', 
+'Coding like a beast',
+'2020-05-20 19:34:43', 
+'2020-05-29 23:35:45', 
+'2020-06-24 16:00:46', 
+ 1
+ );
+ SELECT * FROM task ORDER BY id DESC LIMIT 1;
+ 
+ -- #1b: Change the title of a task:
+UPDATE
+    task
+SET
+    title = 'Tickets to paradise Hotel available'
+WHERE
+    id = 34;
+ SELECT * FROM task WHERE id = 34;
+ 
+ -- #1c Change a task due date:
+UPDATE
+    task
+SET
+    due_date = '2020-12-31 23:59:00'
+WHERE
+    status_id = 3 AND status_id = 2;
+ 
+ SELECT*FROM task WHERE status_id IN (3 , 2);
+ 
+-- #1d Change a task status:
+UPDATE
+    task 
+SET
+    status_id = 3
+WHERE
+    id = 35;
+    
+ SELECT * FROM task;  
+ 
+ -- #1e Mark a task as complete:
+  UPDATE
+    task
+ SET
+    status_id = 1
+WHERE 
+     id = 36;
+SELECT * FROM task;
+    
+-- #1f Delete a task:
+DELETE FROM task 
+WHERE
+    id = 1;
+SELECT * FROM task; 
+ 
+-- -------------------------------------------------------- END OF PART ONE --------------------------------------------------------------------------------------
+
+-- PART 2: School database 
+
+-- #2a Class: with the columns: id, name, begins (date), ends (date):
+CREATE TABLE class(
+   `id` INT AUTO_INCREMENT,
+   `class_name` VARCHAR(255),
+   `start_date` DATE,
+   `end_date` DATE,
+   PRIMARY KEY(id)
+);
+
+-- #2b Student: with the columns: id, name, email, phone, class_id (foreign key):
+CREATE TABLE student(
+   `id` INT AUTO_INCREMENT,
+   `student_name` VARCHAR(255),
+   `email` VARCHAR(100),
+   `phone` INT(10),
+   `class_id` INT(10),
+FOREIGN KEY (`class_id`) REFERENCES `student` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+PRIMARY KEY(id)
+);
+
+-- #2c Create an index on the name column of the student table:
+CREATE INDEX full_name ON student(student_name);
+Show indexes from student;
+
+-- #2d Add a new column to the class table named status which can only have the following values:
+--     not-started, ongoing, finished (hint: enumerations).
+ALTER TABLE class
+ADD COLUMN status VARCHAR(255) AFTER end_date;
```

Comment:
This one is almost correct. You are just using a VARCHAR instead of an ENUM. https://dev.mysql.com/doc/refman/8.0/en/enum.html